flake.nix

```diff
@@ -35,5 +35,17 @@
           drv = packages.rnix-lsp;
         };
         defaultApp = apps.rnix-lsp;
-      });
+      })
+      //
+      rec {
+        overlays = { 
+          default = final: prev: rec {
+          rnix-lsp = self.packages.${prev.system}.rnix-lsp;
+        };
+          nightly = final: prev: rec {
+          rnix-lsp-nightly = self.packages.${prev.system}.rnix-lsp;
```

Comment:
Isn't this rather error-prone to not use `final`/`prev` here, but another nixpkgs? I'd suggest to define the package in an overlay and expose the package *from* the overlay in `self.packages` actually.

Reply:
> Isn't this rather error-prone to not use final/prev here, but another nixpkgs?

Oh yes, you are right. I assumed that `nixpkgs` would be specified in the `follows` attribute, but It makes sense that I shouldn't!


>  I'd suggest to define the package in an overlay and expose the package from the overlay in self.packages actually.

Oh, that is quite neat, I will try that! My first thought was to write a function that we expose at the `overlay` and the `self.packages`.

Reply:
### Proposal: proto-derivations

I've been using this approach as a way to expose "proto-derivations" (the callPackage design pattern where you have a function that when given arguments+evaluated will produce a derivation). This is then a very re-usable thing that avoids many of the problems with overlays, but can still be utilized if so desired.  Other consumers can use `some-flake#proto.default` in any way they wish.
```
{
   outputs = {self,...}: {
     packages.<system>.default = nixpkgs.legacyPackages.<system>.callPackage self.proto.default {inherit naersk-lib;};
     packages.<system>.optimized = nixpkgs.legacyPackages.<system>.callPackage self.proto.default {
        optimized = true;
        inherit naersk-lib;
     };
     overlay.default = final: prev: final.callPackage self.proto.default {};

     # proposing a standard that we expose "proto-derivations" like this
     proto.default = {stdenv, lib, cargo, naerst-lib, optimized ? false }: naersk-lib.buildPackage {
       pname = "asdfas";
     };
   }
}
```
@thufschmitt thoughts about having this be a part of the output schema?